core/src/main/java/org/renjin/primitives/text/Text.java

```diff
@@ -203,6 +203,25 @@ public static void bindtextdomain(String domain, String dirname) {
     // translation not yet supported.
   }
 
+  @Internal
+  public static StringVector enc2utf8(SEXP inputVector) {
+    if (!(inputVector instanceof StringVector)) {
+      throw new EvalException("Error in enc2utf8(" + inputVector + ") : argumemt is not a character vector");
+    }
+
+    StringVector.Builder outputVector = new StringVector.Builder();
+    if (inputVector.length() > 1) {
+      for (int i = 0; i < inputVector.length(); i++) {
+        outputVector.add(((StringVector) inputVector).getElementAsObject(i));

```

Comment:
You will rarely want to use `getElementAsObject()`: prefer the typed methods like `getElementAsString()`. By adding an untyped element here, you are requiring runtime checks to see how it should be handled.